### PR TITLE
LibJS: String.prototype.substring

### PR DESCRIPTION
**Pull request by kessejones**
Added substring function in String.prototype.

### CODE REVIEW

**Comment by awesomekling** [comment]
Please use `clang-format` so that your code [fits with the rest of the codebase](https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md).
Also please squash your commits instead of adding bugs and then fixing them. :)

File: Libraries/LibJS/Runtime/StringPrototype.cpp [hunk 1]
```diff
@@ -307,4 +308,61 @@ Value StringPrototype::trim_end(Interpreter& interpreter)
     return trim_string(interpreter, *this_object, TrimMode::Right);
 }
 
+Value StringPrototype::substring(Interpreter& interpreter)
+{
+    auto* this_object = interpreter.this_value().to_object(interpreter.heap());
```
Comment (awesomekling) [inline Libraries/LibJS/Runtime/StringPrototype.cpp:313]: Missing a null check here. `to_object()` may fail due to an exception being thrown.
```suggestion
    auto* this_object = interpreter.this_value().to_object(interpreter.heap());
    if (!this_object)
        return {};
```

**Comment by kessejones** [comment]
I don't know if there is a problem but I squash all the commits. Now I believe it is ready :)

**Comment by awesomekling** [comment]
Okydoky!

**Pull request merged**